### PR TITLE
[spirv] Support other primitive int/float types.

### PR DESCRIPTION
**Pull request by ehsannas**
PTAL @antiagainst

### CODE REVIEW

File: docs/SPIR-V.rst [hunk 1]
```diff
@@ -87,16 +87,18 @@ Normal scalar types
...
-      HLSL               SPIR-V       Capability       Decoration
-================== ================== =========== ====================
-``bool``           ``OpTypeBool``
-``int``            ``OpTypeInt 32 1``
-``uint``/``dword`` ``OpTypeInt 32 0``
-``half``           ``OpTypeFloat 32``             ``RelexedPrecision``
-``float``          ``OpTypeFloat 32``
-``double``         ``OpTypeFloat 64`` ``Float64``
-================== ================== =========== ====================
+====================================== ================== =========== ====================
```
Comment (antiagainst) [inline docs/SPIR-V.rst]: The following section is for minimal precision scalar types.

File: tools/clang/test/CodeGenSPIRV/type.scalar.hlsl [hunk 2]
```diff
@@ -1,38 +1,85 @@
...
+  min10float m10f;
+
+// CHECK-DAG: %double = OpTypeFloat 64
+// CHECK-DAG: %_ptr_Function_double = OpTypePointer Function %double
+  double doublevar;
+
+// These following variables should use the types already defined above.
+  bool1      boolvar1;
+  int1       intvar1;
+  min16int   m16i1;
```
Comment (antiagainst) [inline tools/clang/test/CodeGenSPIRV/type.scalar.hlsl]: Needs to append `1` to the type to create vector of size 1.

File: tools/clang/lib/SPIRV/TypeTranslator.cpp [hunk 3]
```diff
@@ -34,6 +34,31 @@ inline void roundToPow2(uint32_t *val, uint32_t pow2) {
...
 } // anonymous namespace
 
+bool TypeTranslator::isRelaxedPrecisionType(QualType type) {
+  // Primitive types
+  {
+    QualType ty = {};
+    if (isScalarType(type, &ty)) {
+      if (const auto *builtinType = ty->getAs<BuiltinType>()) {
+        const auto kind = builtinType->getKind();
+        return kind == BuiltinType::Short || kind == BuiltinType::UShort ||
```
Comment (antiagainst) [inline tools/clang/lib/SPIRV/TypeTranslator.cpp]: Can use switch

File: docs/SPIR-V.rst (hunk 1 above)
Comment (ehsannas) [reply docs/SPIR-V.rst]: Oops, i missed that.

File: tools/clang/test/CodeGenSPIRV/type.scalar.hlsl (hunk 2 above)
Comment (ehsannas) [reply tools/clang/test/CodeGenSPIRV/type.scalar.hlsl]: Fixed

File: tools/clang/lib/SPIRV/TypeTranslator.cpp (hunk 3 above)
Comment (ehsannas) [reply tools/clang/lib/SPIRV/TypeTranslator.cpp]: Done

**Pull request merged**